src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc

```diff
@@ -161,14 +161,15 @@ std::optional<edm4eic::MutableCluster> CalorimeterClusterRecoCoG::reconstruct(co
   // Used to optionally constrain the cluster eta to those of the contributing hits
   float minHitEta = std::numeric_limits<float>::max();
   float maxHitEta = std::numeric_limits<float>::min();
-  auto time       = pcl.getHits()[0].getTime();
-  auto timeError  = pcl.getHits()[0].getTimeError();
+  auto time      = 0;
+  auto timeError = 0;
   for (unsigned i = 0; i < pcl.getHits().size(); ++i) {
     const auto& hit   = pcl.getHits()[i];
     const auto weight = pcl.getWeights()[i];
     debug("hit energy = {} hit weight: {}", hit.getEnergy(), weight);
     auto energy = hit.getEnergy() * weight;
     totalE += energy;
+    time += (hit.getTime() * energy - time) / totalE;
```

Comment:
Out of curiosity, is there a particular reason to prefer the energy-weighted time over the earliest?

Reply:
I wanted something that is relatively robust against the inclusion of noise hits. If a noise hit gets included in a cluster, I didn't want it to throw off the entire timing determination.

Reply:
If you want something robust that is closer to earliest hit, then I guess we could do a CFD-style timing determination: the time at which we reach 50% of the maximum in a suitably binned histogram. I don't know how you'd code that up...

Reply:
Ahhh gotcha! That makes sense! I don't know how to code up the CFD-style determination either 😅 

I'm happy with the energy-weighted time for now! I was just curious about the rationale for the choice